src/components/split_button/split_button.tsx

```diff
@@ -0,0 +1,289 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { CommonProps } from '../common';
+
+import { OuiScreenReaderOnly } from '../accessibility';
+import {
+  OuiSplitButtonControl,
+  OuiSplitButtonControlProps,
+  OuiSplitButtonOption,
+} from './split_button_control';
+import { OuiInputPopover } from '../popover';
+import {
+  OuiContextMenuItem,
+  OuiContextMenuItemLayoutAlignment,
+} from '../context_menu';
+import { keys } from '../../services';
+import { OuiI18n } from '../i18n';
+
+enum ShiftDirection {
+  BACK = 'back',
+  FORWARD = 'forward',
+}
+
+export type OuiSplitButtonProps = CommonProps &
+  Omit<
+    OuiSplitButtonControlProps,
+    'onChange' | 'onClick' | 'options' | 'value'
+  > & {
+    /**
+     * Pass an array of options : string or ReactNode
+     */
+    options?: OuiSplitButtonOption[];
+
+    /**
+     * optional index of options item to mark with checkmark
+     */
+    selectedIndex?: string;
+
+    /**
+     * Classes for the context menu item
+     */
+    itemClassName?: string;
+
+    /**
+     * You must pass an `onChange` function to handle selection of an option item
+     */
+    onChange?: (index: number) => void;
+
+    /**
+     * You must pass `onClick` function to handle click of the Primary button
+     */
+    onClick?: () => void;
+
+    /**
+     * Change to `true` if you want horizontal lines between options.
+     * This is best used when options are multi-line.
+     */
+    hasDividers?: boolean;
+
+    /**
+     * Change `OuiContextMenuItem` layout position of icon
+     */
+    itemLayoutAlign?: OuiContextMenuItemLayoutAlignment;
+
+    /**
+     * Applied to the outermost wrapper (popover)
+     */
+    popoverClassName?: string;
+
+    /**
+     * Controls whether the options are shown. Default: false
+     */
+    isOpen?: boolean;
+  };
+
+export const OuiSplitButton = ({
+  fullWidth,
+  options = [],
+  selectedIndex,
+  hasDividers,
+  itemClassName,
+  onChange,
+  onClick,
+  isOpen: propIsOpen,
+  className,
+  popoverClassName,
+  children,
+  ...rest
+}: OuiSplitButtonProps) => {
+  const itemNodes: Array<HTMLButtonElement | null> = [];
+  const [isOpen, setIsOpen] = useState(propIsOpen || false);
+
+  let selectedIndexInt: number | undefined = Number(selectedIndex);
+  if (Number.isNaN(selectedIndexInt)) selectedIndexInt = undefined;
+
```

Comment:
I understand the use of requestAnimationFrame -- to wait one 'tick' for the popover render.   Why should this loop though?  Shouldn't the focus be a "one ping only" event?

Reply:
Yes, I agree. The current logic seems to be doing things it shouldn't. Using `OuiContextMenu` as reference (another component that wraps around `OuiPanel` in a similar way and adds focus in a similar way), RAF is only called once. `OuiContextMenu` takes control over focus, so it's implementation is a little overkill for this situation, but I think something similar could be implemented for this component:

```tsx
const updateFocus = () => {
  requestAnimationFrame(() => {
    if (selectedIndexInt) {
      focusItemAt(selectedIndexInt);
    } else {
      // If no item is selected, just focus the first item
      focusItemAt(0);
    }
  });
};

const openPopover = () => {
  setIsOpen(true);
  updateFocus();
};
```

Reply:
Way better and also feels like it lines up correctly with the definition of RAF.  
Included.